### PR TITLE
Ignore `UV_CACHE_DIR` in `help` tests

### PR DESCRIPTION
**Pull request by sobolevn**
## Summary

I added a new function to push an extra filter to ignore `\[env: UV_CACHE_DIR=.+\]` line and replace it with an empty value.

I also changed `.editorconfig` file, because you actually need empty lines with spaces. Other option: I can add a default filter to ignore empty lines with spaces and replace them with just empty lines.

Plus, I added `setup-uv` action instead of `curl -LsSf https://astral.sh/uv/install.sh | sh`, not sure if that is planned right now, though!

Closes https://github.com/astral-sh/uv/issues/7542

## Test Plan

I executed this locally, it works!

```
» UV_CACHE_DIR=/tmp cargo nextest run --no-fail-fast help                   
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.22s
------------
 Nextest run ID f8ee686c-2fb9-49da-a657-aeeaf4dbb534 with nextest profile: default
    Starting 13 tests across 89 binaries (1693 tests skipped)
        PASS [   1.289s] uv::help help_flag_subsubcommand
        PASS [   1.295s] uv::help help_with_help
        PASS [   1.301s] uv::help help_flag_subcommand
        PASS [   1.317s] uv::help help_short_flag
        PASS [   1.326s] uv::help help_flag
        PASS [   1.331s] uv::help help_with_no_pager
        PASS [   1.333s] uv::help help_subcommand
        PASS [   1.334s] uv::help help
        PASS [   1.334s] uv::help help_unknown_subsubcommand
        PASS [   1.346s] uv::help help_subsubcommand
        PASS [   1.351s] uv::help help_with_global_option
        PASS [   1.380s] uv::help help_unknown_subcommand
        PASS [   0.132s] uv::help help_with_version
------------
     Summary [   1.422s] 13 tests run: 13 passed, 1693 skipped
                                       
```

### CODE REVIEW

**Comment by sobolevn** [comment]
I reverted the CI change, because I don't quite understand how this works ;)

File: crates/uv/tests/common/mod.rs [hunk 1]
```diff
@@ -405,6 +405,14 @@ impl TestContext {
         }
     }
 
+    /// Ignore `UV_CACHE_DIR` env variable in tests.
+    pub fn ignore_cache_dir(&mut self) {
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:409]: The current naming pattern for for this is `with_filtered_...`, e.g., `TestContext::with_filtered_exe_suffix`

**Comment by zanieb** [comment]
In the CI change, it looks like you dropped the `uv python install` command so tests fail. I think you want something like this:

```yaml
      - uses: astral-sh/setup-uv@v3
        with:
          version: "latest"
          enable-cache: true
      - name: "Install required Python versions"
        run: uv python install
```

**Comment by sobolevn** [comment]
Thanks!

**Review by konstin — APPROVED** [review]
Thank you!

**Pull request merged**